src/unknown_0255E0.c

```diff
@@ -1051,8 +1174,153 @@ void func_8003093C(s32 arg0) {
+                index = temp_v0;
+                x = tempObj->segment.trans.x_position;
+                z = tempObj->segment.trans.z_position;
+            }
+        } else if ((i < 4) && (obj != D_8011D388[i].unk34)) {
+            index = i;
+            x = phi_s3[i].trans.x_position;
+            z = phi_s3[i].trans.z_position;
+        }
+        if (index != -1) {
+            x -= obj->segment.trans.x_position;
+            z -= obj->segment.trans.z_position;
+            if (((x * x) + (z * z)) < obj->unk78f) {
+                phi_a0 = sp44->unkC;
+                phi_v1 = sp44->unkE;
+                temp_a3 = sp44->unk9;
+                temp_t0 = sp44->unkA;
+                temp_t1 = sp44->unkB;
+                temp_a1 = sp44->unk10;
+                if (phi_v1 < phi_a0) {
+                    phi_v1 = sp44->unkC;
+                    phi_a0 = sp44->unkE;
+                }
+                if (phi_v1 >= 0x400) {
+                    phi_v1 = 0x3FF;
+                }
+                if (phi_a0 >= phi_v1 - 5) {
+                    phi_a0 = phi_v1 - 5;
+                }
+                temp_v0_3 = &D_8011D388[index];
+                temp_v0_3->unk28 = temp_a3;
+                temp_v0_3->unk29 = temp_t0;
+                temp_v0_3->unk2A = temp_t1;
+                temp_v0_3->unk2C = phi_a0;
+                temp_v0_3->unk2E = phi_v1;
+                temp_v0_3->unk14 = ((temp_a3 << 16) - temp_v0_3->unk0) / temp_a1;
+                temp_v0_3->unk18 = ((temp_t0 << 16) - temp_v0_3->unk4) / temp_a1;
+                temp_v0_3->unk1C = ((temp_t1 << 16) - temp_v0_3->unk8) / temp_a1;
+                temp_v0_3->unk20 = ((phi_a0 << 16) - temp_v0_3->unkC) / temp_a1;
+                temp_v0_3->unk24 = ((phi_v1 << 16) - temp_v0_3->unk10) / temp_a1;
+                temp_v0_3->unk30 = temp_a1;
+                temp_v0_3->unk34 = obj;
+            }
+        }
+    }
+}
+
+#else
 GLOBAL_ASM("asm/non_matchings/unknown_0255E0/obj_loop_fogchanger.s")
+#endif
+
+#ifdef NON_EQUIVALENT
+void func_80030DE0(s32 arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, s32 arg5, s32 arg6) {
+    s32 max, min;
+    unk8011D388 *entry;
+
+    entry = &D_8011D388[arg0];
+    max = arg5;
+    min = arg4;
+    if (arg5 < arg4) {
+        max = arg4;
+        min = arg5;
+    }
+    if (max > 0x3FF) {
+        max = 0x3FF;
+    }
+    if (min >= max - 5) {
+        min = max - 5;
+    }
+    entry->unk28 = arg1;
+    entry->unk29 = arg2;
+    entry->unk2A = arg3;
+    entry->unk2C = min;
+    entry->unk2E = max;
+    entry->unk14 = ((arg1 << 16) - entry->unk0) / arg6;
+    entry->unk18 = ((arg2 << 16) - entry->unk4) / arg6;
+    entry->unk1C = ((arg3 << 16) - entry->unk8) / arg6;
+    entry->unk20 = ((min << 16) - entry->unkC) / arg6;
+    entry->unk24 = ((max << 16) - entry->unk10) / arg6;
+    entry->unk30 = arg6;
+    entry->unk34 = 0;
+}
+#else
 GLOBAL_ASM("asm/non_matchings/unknown_0255E0/func_80030DE0.s")
+#endif
+
+// Unused
+void func_80030FA0(void) {
```

Comment:
Very minor nitpick. I've been marking funcs as UNUSED with a macro lately. Like `UNUSED void func_80030FA0(void) {` . Think I picked it up form Fazana? I have a vague memory of it being useful when doing custom edits to remove unused code so it can be reused elsewhere. Not sure if that's right or wrong, just thought I'd mention it while I was reviewing.